contracts/LimitsChecker.sol

```diff
@@ -0,0 +1,267 @@
+    {
+        return (
+            spent,
+            limit - spent,
+            _getPeriodStartFromTimestamp(currentPeriodEnd - 1),
+            currentPeriodEnd
+        );
+    }
+
+    // ------------------
+    // PRIVATE METHODS
+    // ------------------
+
+    /// currentPeriodEnd is calculated as a calendar date of the beginning of a next month, bi-months, quarter, half year, or year period.
+    /// If, for example, periodDurationMonth = 3, then it is considered that the date changes once a quarter.
+    /// And can take values 01.01, 01.04, 01.07, 01.10.
+    /// Then at the moment when it is necessary to shift the currentPeriodEnd (the condition is fulfilled: block.timestamp >= currentPeriodEnd),
+    /// currentPeriodEnd takes on a new value and spent is set to zero. Thus begins a new period.
+    function _checkAndUpdateLimitParameters() internal {
+        _checkPeriodDurationMonth(periodDurationMonth);
+        if (block.timestamp >= currentPeriodEnd) {
+            currentPeriodEnd = uint128(_getPeriodEndFromTimestamp(block.timestamp));
+            spent = 0;
+        }
+    }
+
+    function _checkPeriodDurationMonth(uint256 _periodDurationMonth) internal view {
+        require(
+            _periodDurationMonth == 1 ||
+                _periodDurationMonth == 2 ||
+                _periodDurationMonth == 3 ||
+                _periodDurationMonth == 6 ||
+                _periodDurationMonth == 12,
+            ERROR_WRONG_PERIOD_DURATION
+        );
+    }
+
+    function _checkLimit(uint256 _payoutSum) internal view {
+        require(_payoutSum <= limit - spent, ERROR_SUM_EXCEEDS_LIMIT);
+    }
+
+    function _increaseSpent(uint256 _payoutSum) internal {
+        spent += uint128(_payoutSum);
+    }
+
+    function _getPeriodStartFromTimestamp(uint256 _timestamp) internal view returns (uint256) {
+        // Get year and number of month of the timestamp:
+        (uint256 _year, uint256 _month, ) = bokkyPooBahsDateTimeContract.timestampToDate(
+            _timestamp
+        );
+        // We assume that the year will remain the same,
+        // because the beginning of the current calendar period will necessarily be in the same year.
+        uint256 _periodStartYear = _year;
+        // Get the number of the start date month:
+        uint256 _periodStartMonth = _getFirstMonthInPeriodFromCurrentMonth(_month);
+        //The beginning of the period always matches the calendar date of the beginning of the month.
+        uint256 _periodStartDay = 1;
+        return
+            bokkyPooBahsDateTimeContract.timestampFromDate(
+                _periodStartYear,
+                _periodStartMonth,
+                _periodStartDay
+            );
+    }
+
+    function _getFirstMonthInPeriodFromCurrentMonth(uint256 _month)
+        internal
+        view
+        returns (uint256 _firstMonthInPeriod)
+    {
+        // To get the number of the first month in the period:
+        //   1. get the number of the period:
```

Comment:
`of the period within the current year starting from its beginning`

Reply:
done!